### PR TITLE
fix-#107: Added footer 

### PR DESCRIPTION
**Pull request by Mohitsen11**
## Summary

This PR adds footer in the homepage.

## Description

This PR adds footer in the homepage as it was not added as per the figma file. I created a footer-layout.tsx file and import it at the end of the homepage. 

## Images

Screenshot (21)

## Issue(s) Addressed

#107

## Prerequisites

- [x] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Couple of observations.
1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
2. It is visible only in the home-page, it should be every where in the entire app ideally

**Comment by Mohitsen11** [comment]
> Couple of observations.
> 
> 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> 2. It is visible only in the home-page, it should be every where in the entire app ideally


I have optimized it and made it visible in the entire app. 
Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.

File: frontend/src/layouts/footer-layout.tsx
Comment (krishnaacharyaa) [inline frontend/src/layouts/footer-layout.tsx:1]: That's nice, I see a new folder created, do we have scope to create header as well? similarly and refactor the code?

**Review by krishnaacharyaa — COMMENTED** [review]
Can you kindly look into the comments

File: frontend/src/pages/add-blog.tsx [hunk 1]
```diff
@@ -122,6 +123,7 @@ function AddBlog() {
   }
 
   return (
+    <Footer>
     <div className="min-h-screen cursor-default bg-slate-50 px-6 py-8 dark:bg-dark">
```
[the remark this answers is not in the document]
Comment (Mohitsen11) [reply frontend/src/pages/add-blog.tsx:127]: Yes, if we follow this approach we have to do this for every page.

Yes, we can have it in a parent level.

[the remark this answers is not in the document]
Comment (krishnaacharyaa) [reply frontend/src/pages/add-blog.tsx:127]: Let's have it at parent level

File: frontend/src/layouts/footer-layout.tsx
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:1]: Yeah, we can do this, header code is written inside the homepage and that can be segregate out as a new header-layout.tsx. That will make the homepage code look clean a bit and in future, we can make changes in the header as well and we can put it where we want.

**Comment by krishnaacharyaa** [comment]
> > Couple of observations.
> > 
> > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> 
> I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> 
> Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.

Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)

Can we have Footer component at the last in the parent component

File: frontend/src/layouts/footer-layout.tsx
Comment (krishnaacharyaa) [reply frontend/src/layouts/footer-layout.tsx:1]: Sure lets' refactor, but lets' not make it wrapper component..

**Comment by Mohitsen11** [comment]
> > > Couple of observations.
> > > 
> > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > 
> > 
> > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> 
> Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> 
> Can we have Footer component at the last in the parent component

Yes, we can have.

File: frontend/src/layouts/footer-layout.tsx
Comment (krishnaacharyaa) [reply frontend/src/layouts/footer-layout.tsx:1]: @Mohitsen11, if you feel it is too much at a shot, we can raise seperator PR for refactoring aswell, I am okay with it

File: frontend/src/pages/add-blog.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mohitsen11) [reply frontend/src/pages/add-blog.tsx:127]: > Let's have it at parent level

Yeah, sure.

File: frontend/src/layouts/footer-layout.tsx
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:1]: Okay, that would be good. Lets have first footer-layout at parent level.

**Comment by Mohitsen11** [comment]
> > > > Couple of observations.
> > > > 
> > > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > > 
> > > 
> > > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> > 
> > 
> > Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> > Can we have Footer component at the last in the parent component
> 
> Yes, we can have.

**Comment by Mohitsen11** [comment]
> > > > > Couple of observations.
> > > > > 
> > > > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > > > 
> > > > 
> > > > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > > > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> > > 
> > > 
> > > Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> > > Can we have Footer component at the last in the parent component
> > 
> > 
> > Yes, we can have.

I've added footer-layout in the parent level.

**Comment by Mohitsen11** [comment]
> > > > > > Couple of observations.
> > > > > > 
> > > > > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > > > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > > > > 
> > > > > 
> > > > > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > > > > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> > > > 
> > > > 
> > > > Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> > > > Can we have Footer component at the last in the parent component
> > > 
> > > 
> > > Yes, we can have.
> 
> I've added footer-layout in the parent level.
And added 'key prop' to the categoryPill component to resolve the error.

Error:- Warning: Each child in a list should have a unique "key" prop.

File: frontend/src/layouts/footer-layout.tsx
Comment (krishnaacharyaa) [inline frontend/src/layouts/footer-layout.tsx:1]: Please optimize for the mobile screen, all the calss without prefix would be for mobile and sm: would be for sm screens and above, you are doing just the reverse kindly change it

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Request for mobile screen optimization and please reduce the height of the footer, it is looking a bit too much in desktop and relative for mobile screen even

File: frontend/src/layouts/footer-layout.tsx
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:1]: I updated the css.

Comment (krishnaacharyaa) [inline frontend/src/layouts/footer-layout.tsx:1]: 1. 
image

We have further scope of optimizing it for mobile screens, Doesn't the font look very big ? Please use 16px , tailwind equivalent for both desktop and mobile.

2.
image
Please make sure they are aligned.

3. Please reduce the height of the footer 
4. I have to scroll to see the footer in both details page and add blogs page, ideally I shouldn't have to scroll as there is much space which is empty which can be effectively utilized. You might have to change viewport related classes

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Thank you for the quick updates Mohit, sorry I am a bit picky when it comes to UI/UX, hope you don't mind, I have noted my observations can you please look into them, feel free to ask questions, so that we don't have to do rework again.

File: frontend/src/layouts/footer-layout.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+function footer() {
+  const newDate = new Date();
+  const year = newDate.getFullYear();
+
+  return (
+      <div className="w-full text-center self-end">
+      <footer className="relative left-0 bottom-0 sm:h-[8vh] flex flex-col sm:flex-row items-center justify-between sm:py-3 sm:px-5 md:px-8 lg:px-20 text-white bg-[#181818]">
+          <section className="text-lg">
+                WonderLust
+          </section>
```
[the remark this answers is not in the document]
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:10]: Correction done!

File: frontend/src/layouts/footer-layout.tsx
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:1]: I have made some changes to meet desired output. I have some question to ask too but before I want feedback on these changes.

File: frontend/src/layouts/footer-layout.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+function footer() {
+  const newDate = new Date();
+  const year = newDate.getFullYear();
+
+  return (
+      <footer className="w-full text-center self-end absolute bottom-0 sm:h-[7vh] flex flex-col sm:flex-row items-center justify-between sm:px-4 md:px-[2rem] lg:px-[4rem] text-white bg-[#181818]">
+          <section className="text-[16px]">
```
Comment (krishnaacharyaa) [inline frontend/src/layouts/footer-layout.tsx:7]: please don't hard code the data, unless there is no other way to solve it. 
1. w.r.t heights use tailwind equivalent of px-2rem something like px-2 or px-1 or px-16 so on
2. use variable for color and define it in tailwind-config.js

File: frontend/src/pages/add-blog.tsx [hunk 4]
```diff
@@ -137,7 +137,7 @@ function AddBlog() {
           </h2>
         </div>
       </div>
-      <div className="flex justify-center ">
+      <div className="flex justify-center mb-[1.5rem]">
         <form onSubmit={handleSubmit} className="md:w-5/6 lg:w-2/3">
```
Comment (krishnaacharyaa) [inline frontend/src/pages/add-blog.tsx:141]: Please remove hard code, and why are we adding this bottom margin?

File: frontend/src/pages/details-page.tsx [hunk 5]
```diff
@@ -56,7 +56,7 @@ export default function DetailsPage() {
         </div>
         <div className="mx-auto flex max-w-4xl flex-col items-center gap-y-4 px-4 py-10">
           <div>
-            <p className="leading-7 text-light-secondary dark:text-dark-secondary md:text-lg">
+            <p className="leading-7 text-light-secondary dark:text-dark-secondary md:text-lg mb-[1rem]">
               {post.description}
```
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:60]: Please remove hard code, and why are we adding this bottom margin? though we have added I see in desktop screen the UI is not as per desired code.
image

File: frontend/src/pages/home-page.tsx [hunk 6]
```diff
@@ -71,7 +71,7 @@ function HomePage() {
       <div className="mx-4 md:mx-8 lg:mx-16">
         <BlogFeed />
         <h1 className="pb-4 text-xl font-semibold dark:text-dark-primary sm:pb-0">All Posts</h1>
-        <div className="flex flex-wrap">
+        <div className="flex flex-wrap mb-[2.8rem]">
           {posts.length === 0
```
Comment (krishnaacharyaa) [inline frontend/src/pages/home-page.tsx:75]: Same here

**Review by krishnaacharyaa — COMMENTED** [review]
Kindly address the comments

File: frontend/src/layouts/footer-layout.tsx (hunk 3 above)
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:7]: Correction done!

File: frontend/src/pages/add-blog.tsx (hunk 4 above)
Comment (Mohitsen11) [reply frontend/src/pages/add-blog.tsx:141]: Done!

File: frontend/src/pages/details-page.tsx (hunk 5 above)
Comment (Mohitsen11) [reply frontend/src/pages/details-page.tsx]: Done!

File: frontend/src/pages/home-page.tsx (hunk 6 above)
Comment (Mohitsen11) [reply frontend/src/pages/home-page.tsx]: Done!

**Comment by Mohitsen11** [comment]
> Kindly address the comments

The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.

**Comment by Mohitsen11** [comment]
> > Kindly address the comments
> 
> The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.



> > Kindly address the comments
> 
> The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.

Um UI is still breaking. I just checked.
Try to resolve it.

**Comment by Mohitsen11** [comment]
> > > Kindly address the comments
> > 
> > 
> > The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.
> 
> > > Kindly address the comments
> > 
> > 
> > The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.
> 
> Um UI is still breaking. I just checked. Try to resolve it.

**Review by krishnaacharyaa — COMMENTED** [review]
Thank you @Mohitsen11 for your efforts :)
As I see... I still see some inconsistency in the implementation
1. 
image
the "All Rights Reserved" text is not aligned to the top blogs. 
2. 
image
Footer should have been properly visible, but because of the logic/hard coded, it isn't 

I don't want to burden you with revisions, may be if you want, I'll do it myself and show the changes later to you, else if you wanna do anyways, kindly test all the corner cases with good quality code(only the required code), for perfection you can test it in your mobile using the private branch i.e
image

Just let me know

File: frontend/src/layouts/footer-layout.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+function footer() {
+  const newDate = new Date();
+  const year = newDate.getFullYear();
+
+  return (
+      <footer className="mt-auto w-full text-center self-end relative bottom-0 min-h-[7vh] sm:h-[7vh] flex flex-col sm:flex-row items-center justify-between py-3 sm:px-4 md:px-8 lg:px-16 text-white bg-dark-footer-background">
+          <section className="text-base">
```
[the remark this answers is not in the document]
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:7]: Correction done!

**Comment by krishnaacharyaa** [comment]
@Mohitsen11 can you leverage flex-1 property, that should ideally solve the issue, min-h-screen looks good to me ...
https://stackoverflow.com/a/75042043/13431819 , see if this helps

**Comment by Mohitsen11** [comment]
> @Mohitsen11 can you leverage flex-1 property, that should ideally solve the issue, min-h-screen looks good to me ... https://stackoverflow.com/a/75042043/13431819 , see if this helps

Okay.

**Comment by krishnaacharyaa** [comment]
@Mohitsen11  Kindly create a bug/issue about "we have to scroll to find the footer even if there is space for the footer in the description, add blog page in both desktop and mobile"
And attach screenshots 
image

Once done, share the ticket here, Will merge this PR

**Comment by Mohitsen11** [comment]
> @Mohitsen11 Kindly create a bug/issue about "we have to scroll to find the footer even if there is space for the footer in the description, add blog page in both desktop and mobile" And attach screenshots image
> 
> Once done, share the ticket here, Will merge this PR

Ticket :- 'https://github.com/krishnaacharyaa/wanderlust/issues/121'

**Review by krishnaacharyaa — APPROVED** [review]
LGTM, Merging PR

File: frontend/src/layouts/footer-layout.tsx
Comment (Mohitsen11) [reply frontend/src/layouts/footer-layout.tsx:1]: > That's nice, I see a new folder created, do we have scope to create header as well? similarly and refactor the code?

I segregated out 'header code' from the 'home-page.tsx' and made a new 'header-layout.tsx' file inside layouts folder. 
Imported all the necessary modules inside 'header-layout.tsx' and added this component in the homepage to keep the UI unchanged.

Want to make a PR.

Comment (krishnaacharyaa) [reply frontend/src/layouts/footer-layout.tsx:1]: Please raise it, let me once have a look
